### DISCUSSION TITLE
Doubts about the Timestamp in dataset

### DISCUSSION THREAD

**Discussion by zbzhu** [PaddlePaddle/PaddleSpatial#39] [BaiduKDDCup2022] [answered]
In the former dataset (__sdwpf_baidukddcup2022_full.CSV__) and the updated dataset (__wtbdata_245days.csv__),  there are totally different values at the same Timestamp for the same turbine.

- data in the former dataset
![image](https://user-images.githubusercontent.com/29559877/167240605-c1f6cf1d-f766-4e5f-883c-8f541b2da50a.png)
- data in the updated dataset
![image](https://user-images.githubusercontent.com/29559877/167240617-80f6a7fb-a47b-48b8-abcf-bdca2621e7a3.png)

So, I have some doubts about the Timestamp in the dataset:
1. Which data should be used in the following phase, which one is more reliable?
2. In the new baseline code, there is a toy testing data, including '0001in.csv' and the corresponding '0001out.csv'. And we should read the data in '0001in.csv' to predict the following 288 data points. However, the starting time in '0001in.csv' is Day=1 and Tmstamp=1. I'm wonder whether the starting time in all testing data would be aligned to Day=1 and Tmstamp=00:00, or it is the continuous timestamp following the provided training data.

**Comment by ethanyxx** [comment, marked as answer]
Hi,
Thanks for your interest in the competition. 
_1. Which data should be used in the following phase, which one is more reliable?_
Reply: The updated dataset should be used in the following phase.

_2. In the new baseline code, there is a toy testing data, including '0001in.csv' and the corresponding '0001out.csv'. And we should read the data in '0001in.csv' to predict the following 288 data points. However, the starting time in '0001in.csv' is Day=1 and Tmstamp=1. I'm wonder whether the starting time in all testing data would be aligned to Day=1 and Tmstamp=00:00, or it is the continuous timestamp following the provided training data._
Reply: The starting time of the testing data is aligned to Day 1 and 00:00, not continuous with the training data. 

Regards,

**Reply by zbzhu** [reply]
Thanks for your reply!

One more thing to confirm.

According to the new report, the stride time step is randomly sampled, hence the starting time of the testing data will be any time in the data. So, **no matter the real starting timestamp of the testing data (e.g., may be 12:30), it will be aligned to 00:00.** That is, we can not use the timestamp as the feature, since the same timestamp in the training data and in the testing data has different meaning.

Best regards.

![image](https://user-images.githubusercontent.com/29559877/167247574-55fc9b50-0225-4751-a4ab-553c161d8c7d.png)

**Comment by ethanyxx** [comment]
Hi,

Thanks for the follow-up questions.

I think there's a bit of misconception here. My previous answer "not continuous with the training data" means that the time sequence of the testing data does not start from the tail of the training data. In other words, the whole testing dataset is selected from a different period of time. However, the definition of Tmstamp is still consistent (not continuous) with the training data. It corresponds to the timestamp (in terms of hours and minutes) of the raw data record regardless of the stride steps. 

Best,
